packages/integration-matomo/src/index.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react'
+
+import {
+  getForegroundColor,
+  IntegrationConfig,
+  createIconComponentFromSimpleIconsSvgPath,
+  useIntegration,
+  useDecision,
+  IntegrationConfigOptions,
+  Tracker,
+  TrackerEvents,
+} from '@techboi/consent-manager'
+
+import Matomo from 'simple-icons/icons/matomo'
+
+declare global {
+  interface Window {
+    _paq?: unknown[]
+  }
+}
+
+interface MatomoTrackerConfig extends IntegrationConfigOptions {
+  matomoURL?: string // @todo these should be required, but TS doesn't like it!
+  siteID?: string // @todo these should be required, but TS doesn't like it!
+  enableLinkTracking?: boolean
+  enableHeartBeatTimer?: boolean
+}
+
+let wasInitialized = false
+
+export const getMatomoTracker = (): TrackerEvents => ({
+  trackEvent: (...args: unknown[]) =>
+    window._paq && window._paq.push(['trackEvent', ...args]),
+  trackPageView: (...args: unknown[]) =>
+    window._paq && window._paq.push(['trackPageView', ...args]),
+  track: (...args: unknown[]) => window._paq && window._paq.push([...args]),
+})
+
+export const useMatomoTracker = ({
+  matomoURL,
+  siteID,
+  enableLinkTracking = true,
+  enableHeartBeatTimer = true,
+}: MatomoTrackerConfig): Tracker => {
+  const [isEnabled] = useDecision('matomo')
+
+  if (!wasInitialized) {
+    const _paq = (window._paq = window._paq || [])
+
+    _paq.push(['trackPageView'])
+    enableLinkTracking && _paq.push(['enableLinkTracking'])
+    enableHeartBeatTimer && _paq.push(['enableHeartBeatTimer'])
+    _paq.push(['setTrackerUrl', `${matomoURL}matomo.php`])
+    _paq.push(['setSiteId', siteID])
+
+    const script = document.createElement('script')
+
+    script.src = `${matomoURL}matomo.js`
+    script.async = true
+
+    document.body.appendChild(script)
+
+    wasInitialized = true
+  }
+
+  const tracker = getMatomoTracker()
+
+  if (!tracker) {
+    throw new Error('Trying to access tracker before it is initialized')
+  }
+
+  return {
+    isEnabled,
```

Comment:
Let's remove this and have the user calling `useDecision` themselves.
Then we can just memoize the `getMatomoTracker` call, so we don't get a new object (causing re-renders) on every call.